sdk/state/ingest.go

```diff
@@ -108,3 +112,33 @@ func (c *Channel) ingestTxToUpdateUnauthorizedCloseAgreement(tx *txnbuild.Transa
 
 	return nil
 }
+
+// ingestTxMetaToUpdateBalances uses the transaction result meta data
+// from a transaction response to update local and remote escrow account
+// balances.
+func (c *Channel) ingestTxMetaToUpdateBalances(resultMetaXDR string) {
+	// If not a valid resultMetaXDR string, return.
+	var txMeta xdr.TransactionMeta
+	err := xdr.SafeUnmarshalBase64(resultMetaXDR, &txMeta)
+	if err != nil {
+		return
+	}
+
+	// Find leder changes for the local and remote escrow accounts, if any,
+	// to update their balance.
+	for _, o := range txMeta.V2.Operations {
+		for _, change := range o.Changes {
+			updated, ok := change.GetUpdated()
+			if !ok {
+				continue
+			}
+
+			switch updated.Data.Account.AccountId.Address() {
+			case c.localEscrowAccount.Address.Address():
+				c.UpdateLocalEscrowAccountBalance(int64(updated.Data.Account.Balance))
+			case c.remoteEscrowAccount.Address.Address():
+				c.UpdateRemoteEscrowAccountBalance(int64(updated.Data.Account.Balance))
+			}
```

Comment:
Does this work for issued assets as well as native balance? We might need to look in a different place for issued assets. You could do that as a separate PR though if you wanted to, might not be trivial.

Reply:
ya  if this looks good I'll extend it to non-native assets, tracked here:
https://github.com/stellar/experimental-payment-channels/issues/237